overlord/state/state_test.go

```diff
@@ -458,6 +458,32 @@ func (ss *stateSuite) TestEnsureBefore(c *C) {
 	c.Check(b.ensureBefore, Equals, 10*time.Second)
 }
 
+func (ss *stateSuite) TestCheckpointPreserveLastIds(c *C) {
+	b := new(fakeStateBackend)
+	st := state.New(b)
+	st.Lock()
+
+	st.NewChange("install", "...")
+	st.NewTask("download", "...")
+	st.NewTask("download", "...")
+
+	// implicit checkpoint
+	st.Unlock()
+
+	c.Assert(b.checkpoints, HasLen, 1)
+
+	buf := bytes.NewBuffer(b.checkpoints[0])
+
+	st2, err := state.ReadState(nil, buf)
+	c.Assert(err, IsNil)
+
+	st2.Lock()
+	defer st2.Unlock()
+
+	c.Assert(st2.NewTask("download", "...").ID(), Equals, "3")
```

Comment:
This is just a personal preference but I like what some companies do when they send you a number. The scheme they use is to prefix it with something short that mentally ties this to the particular product. Perhaps we could use ``"s%d"`` (for snappy) or ``"c%d"`` (for change) or alternatively ``"c-%d"``.